src/operation/mod.rs

```diff
@@ -66,7 +66,8 @@ pub(crate) trait Operation {
     const NAME: &'static str;
 
     /// Returns the command that should be sent to the server as part of this operation.
-    fn build(&self, description: &StreamDescription) -> Result<Command>;
+    /// The operation may store some additional state that is required for handling the response.
+    fn build(&mut self, description: &StreamDescription) -> Result<Command>;
```

Comment:
This was updated to take `&mut self` so that `Insert` could store the `_id`'s it appended to the documents it inserts.

Reply:
It's a pity that so many sites had to be changed to accommodate the single `Insert` case; I wonder if it would be better to keep this as `&self` and cheat with a `RefCell` in `Index`?